CRM/Core/BAO/CustomGroup.php

```diff
@@ -196,7 +196,11 @@ public static function create(&$params) {
     }
     if (!CRM_Utils_System::isNull($extendsChildType)) {
       $b = self::getMungedEntity($params['extends'], $params['extends_entity_column_id'] ?? NULL);
-      $registeredSubTypes = self::getSubTypes()[$b];
+      $subTypes = self::getExtendsEntityColumnValueOptions('validate', ['values' => $params]);
```

Comment:
This is a good step forward because the deprecated function `CRM_Core_BAO_CustomGroup::getSubTypes()` only worked with a hardcoded list of entities while `getExtendsEntityColumnValueOptions()` is metadata-driven.